CorePayments/src/main/java/com/paypal/android/corepayments/api/PatchCCOWithAppSwitchEligibility.kt

```diff
@@ -0,0 +1,76 @@
+package com.paypal.android.corepayments.api
+
+import android.content.Context
+import com.paypal.android.corepayments.APIClientError
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.corepayments.RestClient
+import com.paypal.android.corepayments.common.Headers
+import com.paypal.android.corepayments.graphql.GraphQLClient
+import com.paypal.android.corepayments.graphql.GraphQLResult
+import com.paypal.android.corepayments.model.APIResult
+import com.paypal.android.corepayments.model.ExperimentationContext
+import com.paypal.android.corepayments.model.ExternalResponse
+import com.paypal.android.corepayments.model.PatchCcoWithAppSwitchEligibilityRequest
+import com.paypal.android.corepayments.model.PatchCcoWithAppSwitchEligibilityResponse
+import com.paypal.android.corepayments.model.TokenType
+import com.paypal.android.corepayments.model.Variables
+
+class PatchCCOWithAppSwitchEligibility internal constructor(
+    private val authenticationSecureTokenServiceAPI: AuthenticationSecureTokenServiceAPI,
+    private val graphQLClient: GraphQLClient,
+) {
+
+    constructor(coreConfig: CoreConfig) : this(
+        authenticationSecureTokenServiceAPI = AuthenticationSecureTokenServiceAPI(
+            coreConfig,
+            RestClient(coreConfig)
+        ),
+        graphQLClient = GraphQLClient(coreConfig)
+    )
+
+    suspend operator fun invoke(
+        context: Context,
+        orderId: String,
+        tokenType: TokenType,
+        merchantOptInForAppSwitch: Boolean,
+        paypalNativeAppInstalled: Boolean
+    ): APIResult<PatchCcoWithAppSwitchEligibilityResponse> {
+        return runCatching {
+            val token =
+                when (val tokenResult = authenticationSecureTokenServiceAPI.createLowScopedAccessToken()) {
+                    is APIResult.Success -> tokenResult.data
+                    is APIResult.Failure -> throw tokenResult.error
```

Comment:
qq: why do we need `runCatching` here? It doesn't seem like we're using any Java APIs.

Reply:
not just throwing error programatically, more of safe catching for unknown errors that we didn't thought of, kind of last catch block as fallback

Reply:
I do think we should return an `APIResult` here. It simplifies the API for the caller, they will only need a single `when` statement instead of a combination `when` + `try/catch`.